docs: add concepts for production order, production run and product (#10)
enlyze/enlyze-python: docs/concepts.rst

A *variable* represents a process measure of one :ref:`appliance <appliance>` of which
timeseries data is captured and stored in the ENLYZE platform. One appliance may have
many variables, whereas one variable is only associated with one appliance.

.. _production_order:

Production Order
----------------

A *production order* represents the goal of producing a certain quantity of a given
:ref:`product <product>`. This is how operators know how much of which product to
produce. Production orders are usually created in an external system such as an ERP or
MES and then synchronized into the ENLYZE platform. They are referenced by an identifier

> which oftentimes is a short combination of numbers and/or characters, like FA23000123 or
> 332554.

Author: ```suggestion
which oftentimes is a short combination of numbers and/or characters, like FA23000123.
```

In the ENLYZE platform, a production order always encompasses the production of one

> single :ref:`product <product>` on one single :ref:`appliance <appliance>` within or
> more :ref:`production runs <production_run>`.

Author: ```suggestion
single :ref:`product <product>` on one single :ref:`appliance <appliance>` within one
or more :ref:`production runs <production_run>`.
```

.. _production_run:

Production Run
--------------

> A *production run* is a time frame within a machine was producing a :ref:`product

Author: ```suggestion
A *production run* is a time frame within which a machine was producing a :ref:`product
```

<product>` on an :ref:`appliance <appliance>` in order to complete a :ref:`production
order <production_order>`. A production run always has a beginning and, if it's not
still running, it also has an end.

Usually, the operator of the appliance uses an interface to log the time when a certain
production order has been worked on. For instance, this could be the appliance's HMI or
a tablet computer next to it. In German, this is often referred to as

> "Betriebsdatenerfassung (BDE)".

Author: ```suggestion
*Betriebsdatenerfassung* (BDE). It is common, that a production order is not completed in one go, but is interrupted several times for very different reasons, like a breakdown of the appliance or a public holiday. These interruptions lead to the creation of multiple production runs for a single production order.
```

NB: text will need to be aligned with a proper editor.

.. _product:

Product
-------

A *product* is the output of the production process which is executed by an
:ref:`appliance <appliance>`, driven by a :ref:`production order <production_order>`. In
the real world, an appliance might have some additional outputs, but only the main

> output (the product) modeled in the ENLYZE platform. Similarly to the production order,

Author: ```suggestion
output (the product) is modeled in the ENLYZE platform. Similarly to the production order,
```

> a product is referenced by an identifier originating from a customer system, which is

Author: ```suggestion
a product is referenced by an identifier originating from a customer's system, that gets
```

A *variable* represents a process measure of one :ref:`appliance <appliance>` of which
timeseries data is captured and stored in the ENLYZE platform. One appliance may have
many variables, whereas one variable is only associated with one appliance.

.. _production_order:

> Production Order

Reviewer: Alternatively, we could start describing what shop floor workers do in order to complete a production order: Produce the specified product in the specified quantity until a certain deadline. Because there might be interruptions such as weekends, high priority orders and what not, production orders might be completed within multiple intervals, which we define as *production runs*

Reviewer: (Food for thought, no direct change request)

Author: > Alternatively, we could start describing what shop floor workers do in order to complete a production order: Produce the specified product in the specified quantity until a certain deadline. Because there might be interruptions such as weekends, high priority orders and what not, production orders might be completed within multiple intervals, which we define as production runs

But that's mostly a description of a production run, isn't it? Moreover, we'd rather have to explain *our* concept of a production order, right? And I'd say that a production order has more to do with the person planning the production than the shopfloor workers. But I'll add a sentence into that direction, see the comment below.

----------------

A *production order* represents the goal of producing a certain quantity of a

> given :ref:`product <product>`. Production orders are usually created in an ERP
> system or MES of the customer and then synchronized into the ENLYZE platform.

Author: Agrred. Maybe add "external" then?

```suggestion
given :ref:`product <product>`. Production orders are usually created in an external ERP
system or MES and then synchronized into the ENLYZE platform.
```

> system or MES of the customer and then synchronized into the ENLYZE platform.

Reviewer: :+1:

> system or MES of the customer and then synchronized into the ENLYZE platform.

Author: ```suggestion
A *production order* represents the goal of producing a certain quantity of a
given :ref:`product <product>`. This is how operators are told how much of which product to produce. Production orders are usually created in an ERP
system or MES of the customer and then synchronized into the ENLYZE platform.
```

Reviewer: :+1:

Author: 9dbcdb2

They are referenced by an identifier which oftentimes is a short combination of
numbers and/or characters, like FA23000123 or 332554.

In the ENLYZE platform, a production order always encompasses the production of
one single :ref:`product <product>` on one single :ref:`appliance <appliance>`.

.. _production_run:

Production Run
--------------

> A *production run* represents the real allocation of a :ref:`production order

Reviewer: While technically correct, I'm not sure how easy it is to grasp this.

Reviewer: Maybe something along the lines of:

A production run represents an time frame during which an appliance was working towards completing production order[..]?

Author: Hm, maybe like this?

> A production run represents **a time span when an appliance was allocated to a production order.**
> ....
> A production order oftentimes **is not finished in one go** due to interruptions like a breakdown or a weekend. Each of these individual, continous allocations is called a production run in the ENLYZE platform. **Thus, there can be multiple production runs for one production order.**

Bold sections are my proposed changes.

Reviewer: I'll create a new multi-line suggestion

<production_order>` on an :ref:`appliance <appliance>`. Usually, the operator of
the appliance uses some kind of interface to log the time when a certain

> production order has been worked on. A production order oftentimes is not

Reviewer: :+1:

> the appliance uses some kind of interface to log the time when a certain
> production order has been worked on. A production order oftentimes is not

Author: What about this?

```suggestion
the appliance uses an interface to log the time when a certain
production order has been worked on. For instance, this could be the appliance's HMI or a tablet computer next to it. A production order oftentimes is not
```

allocated continously on an appliance due to interruptions like a breakdown or
weekend. Each of these individual, continous allocations is called a *production
run* in the ENLYZE platform. It always has a beginning and, if it's not still
running, it also has an end. The ENLYZE platform calculates different aggregates
from the timeseries data of the appliance for each production run.

.. _product:

Product
-------

A *product* is the output of the production process which is executed by an
:ref:`appliance <appliance>`, driven by a :ref:`production order
<production_order>`. In the real world, an appliance might have some additonal
outputs, but only the main output (the product) modelled in the ENLYZE platform.
Similarly to the production order, a product is referenced by an identifier

> originating from a system of the customer, which is synchronized into the ENLYZE

Reviewer: ```suggestion
originating from a customer system, which is synchronized into the ENLYZE
```

Author: Haha, I had that before but wasn't sure.

A *variable* represents a process measure of one :ref:`appliance <appliance>` of which
timeseries data is captured and stored in the ENLYZE platform. One appliance may have
many variables, whereas one variable is only associated with one appliance.

.. _production_order:

Production Order
----------------

A *production order* represents the goal of producing a certain quantity of a
given :ref:`product <product>`. This is how operators are told how much of which
product to produce. Production orders are usually created in an external ERP
system or MES and then synchronized into the ENLYZE platform. They are
referenced by an identifier which oftentimes is a short combination of numbers
and/or characters, like FA23000123 or 332554.

In the ENLYZE platform, a production order always encompasses the production of
one single :ref:`product <product>` on one single :ref:`appliance <appliance>`.

.. _production_run:

Production Run
--------------

> A *production run* represents the real allocation of a :ref:`production order
> <production_order>` on an :ref:`appliance <appliance>`. Usually, the operator of

Reviewer: ```suggestion
A production run represents a time interval when an appliance was allocated to a production order. Put differently, the appliance was producing in order to complete the production order
```
